File: factorial/csharp_factorial/factorial.cs
using System;

public class Factorial {
  public static void Main( string[] args ) {
    if( args.Length != 0 ) {
      Console.Error.WriteLine( "Usage: " + Environment.GetCommandLineArgs()[0] + " <noargs>" );
      Environment.Exit( 1 );
    }

    for( uint idx_i = 1; idx_i < 13; idx_i++ ) {
        Console.WriteLine("factorial(" + idx_i + ")=<" + factorial(idx_i) + ">");
    }
  }

  private static uint factorial( uint n ) {
    if( n == 0 ) {
      return 0;
    } else if( n == 1 ) {
      return 1;
    } else {
      return(n * factorial( n - 1 ));
    }
  }
}

File: factorial/fsharp_factorial/factorial.fs
//#light

open System

let rec factorialMatch n =
  match n with
  | 0 -> 0
  | 1 -> 1
  | _ -> n * factorialMatch( n - 1 )

let rec factorial x =
  if x < 1 then 1
  else x * factorial (x - 1)

[<EntryPoint>]
let main args =
  let argv = Environment.GetCommandLineArgs()

  if argv.Length <> 1 then
    eprintfn "Usage: %s <noargs>" argv.[0]
    -1
  else
    for idx_i in 1 .. 12 do
      printfn "factorial(%d)=<%d>" idx_i (factorial idx_i)
    0
